crates/bindings/src/lib.rs

```diff
@@ -71,19 +71,35 @@ impl ReducerContext {
 // #[global_allocator]
 // static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
+struct Buf {
+    vec: UnsafeCell<Vec<u8>>,
+}
+
+impl Buf {
+    const fn new() -> Self {
+        Self {
+            vec: UnsafeCell::new(Vec::new()),
+        }
+    }
+
+    fn with<R>(&self, f: impl FnOnce(&mut Vec<u8>) -> R) -> R {
+        let ptr = self.vec.get();
+        let buf = unsafe { &mut *ptr };
+        f(buf)
+    }
+}
```

Comment:
Oh yeah this is not the api i want, don't worry :)

Reply:
It's the thing where I had 3 minutes to finish typing something before a meeting

Anyway, the motivation here is that there is a comment in the bindings that claims that `// This gets optimized away to a normal global since wasm32 doesn't have threads by default.` which is (sadly) untrue.